Scrabble.java

```diff
@@ -48,26 +48,110 @@ public static void init() {
 
 	// Checks if the given word is in the dictionary.
 	public static boolean isWordInDictionary(String word) {
-		//// Replace the following statement with your code
+		for (int i = 0; i < NUM_OF_WORDS; i++) {
+			if (DICTIONARY[i].equals(word) ) {
+				return true;
+			}
+		}
 		return false;
 	}
 	
 	// Returns the Scrabble score of the given word.
 	// If the length of the word equals the length of the hand, adds 50 points to the score.
 	// If the word includes the sequence "runi", adds 1000 points to the game.
 	public static int wordScore(String word) {
-		//// Replace the following statement with your code
-		return 0;
+		if (word == "" || word == " " || word == null) {
+			return 0;
+		}
+
+		int score = 0;
+		int addLetters = 0;
+
+		boolean[] arrRUNI = new boolean[4];
+		for(int j=0; j<arrRUNI.length; j++ ){
+			arrRUNI[j] = false;
+		}
+
+		for(int i=0; i< word.length(); i++){
+			if (MyString.subsetOf(""+word.charAt(i), "aeilnorstu")) {
+				addLetters++;
+			}else if (MyString.subsetOf(""+word.charAt(i), "dg")) {
+				addLetters+=2;
+			}else if (MyString.subsetOf(""+word.charAt(i), "bcmp")) {
+				addLetters+=3;
+			}else if (MyString.subsetOf(""+word.charAt(i), "fhvyw")) {
+				addLetters+=4;
+			}else if (MyString.subsetOf(""+word.charAt(i), "k")) {
+				addLetters+=5;
+			}else if (MyString.subsetOf(""+word.charAt(i), "jx")) {
+				addLetters+=8;
+			}else if (MyString.subsetOf(""+word.charAt(i), "zq")) {
+				addLetters+=10;
+			}

```

Comment:
1. Why not use the given array of scores?
2. why not use "dg".indexOf(word.charAt(i)) if done this way